examples/generative/corrdiff/training/training_loop.py

```diff
@@ -224,13 +225,16 @@ def training_loop(
                 img_lr = img_lr.to(device).to(torch.float32).contiguous()
                 labels = labels.to(device).contiguous()
 
-                loss = loss_fn(
-                    net=ddp,
-                    img_clean=img_clean,
-                    img_lr=img_lr,
-                    labels=labels,
-                    augment_pipe=augment_pipe,
-                )
+                with torch.autocast(
+                    "cuda", dtype=torch.bfloat16, enabled=(fp_mode == "amp")
```

Comment:
Does it make sense to make the `dtype` configurable, to support `fp16` as well?

Reply:
It's easy to add technically (maybe using `fp_mode == 'amp-fp16'` and `'fp_mode' == 'amp-bf16'`). However I haven't tested AMP with FP16, while I have fully trained CorrDiff with BF16 so I'm confident it'll work ok. Is it better to add it now anyway?

Reply:
No we can keep the code as is. But would be nice to clarify in the README or even as a comment [here](https://github.com/NVIDIA/modulus/blob/b10fe30c8bb18aade11650817731ece8fe95ec48/examples/generative/corrdiff/conf/config_train_base.yaml#L44) that the AMP implementation uses `bfloat16`